lib/posix/spawn/child.rb

```diff
@@ -244,14 +257,27 @@ def read_and_write(input, stdin, stdout, stderr, timeout=nil, max=nil)
 
           # read from stdout and stderr streams
           ready[0].each do |fd|
-            buf = (fd == stdout) ? @out : @err
+            chunk = nil
             begin
-              buf << fd.readpartial(BUFSIZE)
+              chunk = fd.readpartial(BUFSIZE)
             rescue Errno::EAGAIN, Errno::EINTR
             rescue EOFError
               readers.delete(fd)
               fd.close
             end
+
+            abort = false
+            if chunk
+              if fd == stdout
+                abort = (@stdout_buffer.write(chunk) == 0)
```

Comment:
I don't feel like this is a safe way to test for aborting the operation. The output object could be simply refusing to write the current chunk but is not done consuming the stream.
Why not use an exception for this test instead? If the consumer raises `Posix::Spawn::Aborted` then we clearly know to abort.